### PR TITLE
falgas_m89.cpp: Dumped and added "Rally Montecarlo" for Falgas M89 hw

### PR DESCRIPTION
**Pull request by clawgrip**
The game is a modified version of Konami Hyper Rally for MSX.
Extends the Falgas M89 hardware for kiddie rides with an external video board.

New NOT_WORKING machine
--------------------------------
Rally Montecarlo [Museo del Recreativo]

### CODE REVIEW

File: src/mame/misc/falgas_m89.cpp [hunk 1]
```diff
@@ -76,6 +112,7 @@ class falgasm89_state : public driver_device
 	void io_map(address_map &map);
 
 	required_device<i8085a_cpu_device> m_maincpu;
+	optional_device<i8085a_cpu_device> m_videocpu;
```
[the remark this answers is not in the document]
Comment (clawgrip) [reply src/mame/misc/falgas_m89.cpp:115]: Sure! Can you please point to a good example on how to do it?

[the remark this answers is not in the document]
Comment (clawgrip) [reply src/mame/misc/falgas_m89.cpp]: Done!

File: src/mame/misc/falgas_m89.cpp [hunk 2]
```diff
@@ -41,13 +41,46 @@
...
 
 #include "emu.h"
 #include "cpu/i8085/i8085.h"
-//#include "machine/i8255.h"
+#include "machine/i8155.h"
+#include "machine/i8255.h"
+#include "screen.h"
 #include "sound/ay8910.h"
 #include "speaker.h"
+#include "video/tms9928a.h"
```
[the remark this answers is not in the document]
Comment (clawgrip) [reply src/mame/misc/falgas_m89.cpp]: Done!

File: src/mame/misc/falgas_m89.cpp [hunk 3]
```diff
@@ -151,16 +201,52 @@ void falgasm89_state::cbully(machine_config &config)
 	psg.port_b_read_callback().set(FUNC(falgasm89_state::psg_pb_r));
 }
 
+void falgasm89_state::falgasm89(machine_config &config)
+{
+	falgasm89_simple(config);
+
+	I8255(config, "i8255"); // NEC D71055C
+}
```
Comment (cuavas) [inline src/mame/misc/falgas_m89.cpp:209]: What actually uses this configuration?  As far as I can tell, it isn’t used by anything right now.  Does the declaration need to be marked as unused?  What’s the 8255 actually used for when it’s present?  More I/O?

File: src/mame/misc/falgas_m89.cpp [hunk 4]
```diff
@@ -151,16 +201,52 @@ void falgasm89_state::cbully(machine_config &config)
...
+
+void falgasm89_video_state::falgasm89_video(machine_config &config)
+{
+	falgasm89_state::falgasm89(config);
```
Comment (cuavas) [inline src/mame/misc/falgas_m89.cpp:213]: You don’t need to put the base class name qualifier here – there’s no identically named member in the derived class.

File: src/mame/misc/falgas_m89.cpp (hunk 3 above)
Comment (cuavas) [reply src/mame/misc/falgas_m89.cpp:209]: Oh my bad, the 8255 is used with the video version.  Are there actually games that use the 8255 without the video hardware, or does the 8255 only need to be populated to communicate with the video hardware?

File: src/mame/misc/falgas_m89.cpp (hunk 4 above)
Comment (clawgrip) [reply src/mame/misc/falgas_m89.cpp]: Done!

File: src/mame/misc/falgas_m89.cpp (hunk 3 above)
Comment (clawgrip) [reply src/mame/misc/falgas_m89.cpp:209]: It's hard to say for sure... There are about seven machines using M89 hardware, and just one with video. I'd say that yes, there are other machines without video that use the 8255 (more buttons, sensors, etc.).

**Comment by cuavas** [comment]
This driver file can probably be moved to skeleton now that there isn’t an artificial separation of arcade projects, but that’s something to do when a whole lot of moves are going to be done at once.

**Pull request merged**